docs/application/native/guides/ui/efl/wearable/component-gengrid.md

```diff
@@ -0,0 +1,222 @@
+  - To add an item to the beginning of the grid, use the similar `elm_gengrid_item_prepend()` function.
+  - To insert an item before the indicated item, use the `elm_gengrid_item_insert_before()` function.
+  - To insert an item after the indicated item, use the `elm_gengrid_item_insert_after()` function.
+
+- Set the orientation.
+
+  A gengrid can display its items using a horizontal or vertical layout. In the horizontal layout, the items are displayed in columns from top to bottom, starting a new column when the space for the current column is filled. In the vertical layout, items are set in rows from left to right. By default, the vertical layout is used. To set the horizontal layout:
+
+  ```
+  elm_gengrid_horizontal_set(gengrid, EINA_TRUE);
+  ```
+
+- Delete items and clear the grid:
+
+  - To delete a single gengrid item, use the `elm_object_item_del()` function.
+  - To clear the grid and delete all items, use the `elm_gengrid_clear()` function.
+
+  If a gengrid object is deleted by the `evas_object_del()` function, all items are deleted automatically. On item deletion, the `del()` function in the item class is called.
+
+- Update items.
+
+  If the content of an item changes, use the `elm_gengrid_item_update()` function for the gengrid to update the item. The gengrid re-realizes the item and calls the functions in the `_Elm_Gengrid_Item_Class` class for it.
+
+  You can also use the `elm_gengrid_item_fields_update()` function to update only specific parts:
+
+  - `ELM_GENGRID_ITEM_FIELD_TEXT`
+  - `ELM_GENGRID_ITEM_FIELD_CONTENT`
+  - `ELM_GENGRID_ITEM_FIELD_STATE`
+
+- Select or disable items.
+
+  To manually select an item, use the `elm_gengrid_item_selected_set()` function. To select multiple items, you must enable the multi-selection mode:
+
+  ```
+  elm_gengrid_multi_select_set(gengrid, EINA_TRUE);
+  ```
+
+  To retrieve the selected item, use the `elm_gengrid_selected_item_get()` function. When the multi-selection mode is enabled, you can retrieve the selected items with the `elm_gengrid_selected_items_get()` function, which returns a list of all selected items.
+
+  To disable an item manually, use the `elm_object_item_disabled_set()` function.
+
+## Styles
+
+A gengrid item can have 0 or more texts, 0 or more contents, and 0 or more boolean states. The number of texts and contents depends on the Edje item theme style used for the gengrid items. In the `default` Tizen gengrid item theme style, the items can have 1 content parts that can be set with the `elm.swallow.icon` part names.
+
+The following gengrid styles and related item styles are supported:
+
+- `default`
+  - `default`
+
+
+**Figure: Gengrid styles**
+
+![Gengrid styles](./media/gengrid_styles.png)
+
+The following table provides more information on the available gengrid item styles.
+
+**Table: Gengrid item styles**
+
+| Style                                    | Sample                                   | Text part  | Swallow part                        |
```

Comment:
I see that the text part column is empty, is that intentional?

Reply:
Yes.
as wearable device is too small to show the text with icon in grid view, 
so we only have swallow part for wearable.

Reply:
If "Text part" is not supported, how about removing the "Text part" column, or describing it with adding a sentence as like "not supported" to the cell?

Reply:
adding not supported in the text part cell.
as there will be a chance to adding new styles, leave text part column is more clear to viewer.
thanks for good opinion.